resolver/rrcache.go

```diff
@@ -85,6 +85,12 @@ func (rrCache *RRCache) Clean(minExpires uint32) {
 		lowestTTL = maxTTL
 	}
 
+	// Adjust return code if there are no answers
+	if rrCache.RCode == dns.RcodeSuccess &&
+		len(rrCache.Answer) == 0 {
+		rrCache.RCode = dns.RcodeNameError
+	}
+
```

Comment:
Should we document when this can actually happen? Because AFAIK we return an error if the response gets empty because we removed all out-of-scope answers. Is there any other situation we can cause this (except from an invalid upstream-answer)?